src/record/record.js

```diff
@@ -308,6 +309,9 @@ Record.prototype._$onMessage = function( message ) {
 	else if( message.data[ 0 ] === C.EVENT.MESSAGE_DENIED ) {
 		clearInterval( this._readAckTimeout );
 		clearInterval( this._readTimeout );
+	} else if( message.action === C.ACTIONS.SUBSCRIPTION_HAS_PROVIDER ) {
+		this.hasProvider = message.data[ 1 ];
+		this.emit( 'hasProviderChanged', message.data[ 1 ] );
```

Comment:
It would be inconsistent with other events (within the record class), like 
- ready
- discard
- delete
- error